### PR TITLE
FR-9202 - remove frontegg's proxy middlewares

### PR DESCRIPTION
**Pull request by eran-frontegg**
BREAKING CHANGE: support for outdated "frontegg"/"fronteggNextJs" middlewares is now discontinued

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -1,174 +1,82 @@
...
-
-![alt text](https://fronteggstuff.blob.core.windows.net/frongegg-logos/logo-transparent.png)
-
-Frontegg is a web platform where SaaS companies can set up their fully managed, scalable and brand aware - SaaS features and integrate them into their SaaS portals in up to 5 lines of code.
-
+<br />
+<div align="center">
+<img src="https://fronteggstuff.blob.core.windows.net/frongegg-logos/logo-transparent.png" alt="Frontegg Logo" width="400" height="90">
+
+<h3 align="center">Frontegg Node.js client</h3>
```
Comment (raz-shlomo-frontegg) [inline readme.md:5]: Client

File: readme.md [hunk 2]
```diff
@@ -1,174 +1,82 @@
...
+    ·
+    <a href="https://github.com/frontegg/nodejs-sdk/issues">Request Feature</a>
+  </p>
+</div>
+
+<h3>Table of Contents</h3>
+<ul>
+    <li><a href="#installation">Installation</a></li>
+    <li><a href="#usage">Usage</a></li>
+    <li><a href="#breaking-changes">Events Client upgrade notice</a></li>
```
Comment (raz-shlomo-frontegg) [inline readme.md:25]: link

File: src/components/logger/index.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import * as winston from 'winston';
 
-const logger = winston.createLogger({
+const index = winston.createLogger({
```
Comment (raz-shlomo-frontegg) [inline src/components/logger/index.ts:3]: index

**Review by raz-shlomo-frontegg — APPROVED** [review]

File: index.ts [hunk 4]
```diff
@@ -1,33 +1,26 @@
...
-import { FronteggPermissions } from './src/permissions';
-import { RbacMiddleware } from './src/rbac';
-import { SsoClient } from './src/sso';
-import { TenantsClient } from './src/tenants';
-import { AuthzClient } from './src/authz';
-import { HttpClient } from './src/http';
+import { ContextHolder } from './src/components/context';
+import { withAuthentication } from './src/middlewares';
+import {
+  AuditsClient,
```
Comment (eldad-frontegg) [inline index.ts:5]: notification delete, sso is more for the proxy. identity client do not export. AuthzClient check about it? same for tenants and audtits.

File: index.ts [hunk 5]
```diff
@@ -1,33 +1,26 @@
...
+  NotificationsClient,
+  TenantsClient,
+  SsoClient,
+  IdentityClient,
+  AuthzClient,
+  HttpClient,
+} from './src/clients';
 
-export * from './src/events';
+export * from './src/clients/events';
```
Comment (eldad-frontegg) [inline index.ts:14]: check what is the usage here and if customers are using it. I know it was used for the connectivity feature which is currently deprecated and maybe for the webhook feature

File: index.ts [hunk 6]
```diff
@@ -1,33 +1,26 @@
...
 
 export {
   AuditsClient,
   ContextHolder,
   NotificationsClient,
   TenantsClient,
-  frontegg,
-  fronteggNextJs,
```
Comment (eldad-frontegg) [inline index.ts:22]: check with @aviadmizrachi or @davidantoon

File: package.json [hunk 7]
```diff
@@ -22,15 +22,13 @@
   "dependencies": {
     "@slack/web-api": "^6.7.2",
```
Comment (eldad-frontegg) [inline package.json:23]: if you remove the events you might want to remove that as well

File: src/middlewares/with-authentication.ts [hunk 8]
```diff
@@ -41,7 +42,8 @@ export function withAuthentication({ roles = [], permissions = [] }: IWithAuthen
       user = await IdentityClient.getInstance().validateIdentityOnToken(token, { roles, permissions });
     } catch (e) {
       const { statusCode, message } = e;
-      res.status(statusCode).send(message);
+      Logger.debug(message);
```
Comment (eldad-frontegg) [inline src/middlewares/with-authentication.ts:45]: .error

File: src/types/express/index.d.ts [hunk 9]
```diff
@@ -0,0 +1,11 @@
+import { IUser } from '../../middlewares';
+
+export {};
+
+declare global {
+  namespace Express {
+    export interface Request {
+      user?: IUser;
```
Comment (eldad-frontegg) [inline src/types/express/index.d.ts:8]: maybe lets change it to frontegg {user}

File: index.ts (hunk 4 above)
Comment (eran-frontegg) [reply index.ts]: For now keeping tenants and audits clients

File: index.ts (hunk 6 above)
Comment (eran-frontegg) [reply index.ts:22]: David said they transferred the fronteggNextJs code to their own repository, so this can be deleted

**Review by eldad-frontegg — APPROVED** [review]

**Pull request merged**